wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/RecipeFileContentProvider.java

```diff
@@ -20,16 +21,18 @@
  * @author Max Shaposhnyk
  */
 public interface RecipeFileContentProvider {
+  RecipeFileContentProvider NullProvider = name -> null;
 
   /**
    * Fetches content of the file specified under 'local' field fo recipe-type tools in {@link
    * org.eclipse.che.api.devfile.model.Devfile}.
    *
-   * @param localFileName file name to fetch content. Only devfile-relative files are currently
+   * @param localFileName file name to fetch content. Only devfile-related files are currently
    *     supported, so it means file should be localed at the same directory level as devfile (no
    *     matter in repository or PR or branch etc )
-   * @return content of the specified file
+   * @return content of the specified file or null if content is not found
```

Comment:
Not sure if we need these changes in `interface` since it already defines that it throws an exception if a content retrieving issue happens.
Also `NullProvider` is used only in one Test class.

Could you please elaborate more, why we need it?

Reply:
Actually, we have code that returns null if there is no file requested. See https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/URLFactoryBuilder.java#L110
So, it is not true that exception would be thrown in such a case. That's why I added nullable annotation and changed javadocs.

As to the NullProvider it is pure syntax sugar which is indeed used just in one place. If you think that we don't need it I can easily remove NullProvider

Reply:
I think it's a bug that `URLFactoryBuilder` return null, because user will see a confusing message like `content is empty` while it is internal server exception that content provider is missing.

Reply:
Fixed